cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -25,10 +25,22 @@ predicate memberMayBeVarSize(Class c, MemberVariable v) {
 }
 
 /**
- * Get the size in bytes of the buffer pointed to by an expression (if this can be determined).
+ * Gets the expression associated with `n`. Unlike `n.asExpr()` this also gets the
+ * expression underlying an indirect dataflow node.
  */
-language[monotonicAggregates]
-int getBufferSize(Expr bufferExpr, Element why) {
+private Expr getExpr(DataFlow::Node n) {
+  result = [n.asExpr(), n.asIndirectArgument(), n.asIndirectArgument()]
```

Comment:
```suggestion
  result = [n.asExpr(), n.asIndirectArgument()]
```
or is there perhaps another indirect dataflow node besides indirect arguments?

Reply:
@rdmarsh2 ^?

Reply:
Good point, I noticed that after approving and thought I dismissed my review.

Reply:
Thanks for noticing this @intrigus-lgtm! I've been away on vacation last week, but I'll take a look at this. I _think_ the `asIndirectArgument` should actually have been `asIndirectExpr`, but I need to page back in all the context later today 😅.

Reply:
Fixed in a7d6f4ff19e92c814d83d40f906752a9ed591ad8. Indeed, `asIndirectExpr` was the right predicate to use.